packages/nc-gui/components/virtual-cell/barcode/JsBarcodeWrapper.vue

```diff
@@ -0,0 +1,38 @@
+<script lang="ts" setup>
+import JsBarcode from 'jsbarcode'
+
+const props = defineProps({
+  barcodeValue: { type: String, required: true },
+  barcodeFormat: { type: String, required: true },
+})
+
+const emit = defineEmits(['onClickBarcode'])
+const barcodeSvgRef = ref(null)
+const errorForCurrentInput = ref(false)
+
+const generate = () => {
+  try {
+    JsBarcode(barcodeSvgRef.value, String(props.barcodeValue), {
+      format: props.barcodeFormat,
+    })
+    errorForCurrentInput.value = false
+  } catch (e) {
+    console.log('e', e)
+    errorForCurrentInput.value = true
+  }
+}
+
+const onBarcodeClick = (ev: MouseEvent) => {
+  ev.stopPropagation()
+  emit('onClickBarcode')
+}
+
+watch(() => props.barcodeValue, generate)
+watch(() => props.barcodeFormat, generate)
```

Comment:
watch API can take multiple sources with the single handler. You can combine them to

```ts
watch([() => props.barcodeValue, () => props.barcodeFormat], generate)
```

Reply:
Thank you, it's a good hint:)